### ISSUE TITLE
图像推理效果不佳

### ISSUE THREAD

**Issue by meimeizeng** [LabShuHangGU/CTMSR#12]
作者您好，我利用您发布的预训练模型在imagenet测试集图像上进行测试，但是得到的可视化结果过曝光。请问是什么原因，我似乎没有修改任何test.yml文件的任何参数。# GENERATE TIME: Thu Jul 25 02:21:52 2024

# general settings
name: ctmsr_test
model_type: EdmUNetRealModel
scale: 4
num_gpu: 1
manual_seed: 0
find_unused_parameters: False

# dataset and data loader settings

degradation:
  sf: 4
  # the first degradation process
  resize_prob: [0.2, 0.7, 0.1]  # up, down, keep
  resize_range: [0.15, 1.5]
  gaussian_noise_prob: 0.5
  noise_range: [1, 30]
  poisson_scale_range: [0.05, 3.0]
  gray_noise_prob: 0.4
  jpeg_range: [30, 95]

  # the second degradation process
  second_order_prob: 0.5
  second_blur_prob: 0.8
  resize_prob2: [0.3, 0.4, 0.3]  # up, down, keep
  resize_range2: [0.3, 1.2]
  gaussian_noise_prob2: 0.5
  noise_range2: [1, 25]
  poisson_scale_range2: [0.05, 2.5]
  gray_noise_prob2: 0.4
  jpeg_range2: [30, 95]

  gt_size: 256  #
  resize_back: true
  use_sharp: False

datasets:

  val_1:
    name: ImageNet
    type: PairedImageDataset
    dataroot_gt: /root/autodl-tmp/data/ImageNet_SR/train/HR #./datasets/imagenet256/gt
    dataroot_lq: /root/autodl-tmp/data/ImageNet_SR/train/LR_bicubic/X4 #./datasets/imagenet256/lq_real
    mean: 0.5
    std: 0.5
    gt_size: 256
    io_backend:
      type: disk

  val_2:
      name: RealSet65
      type: SingleImageDataset
      dataroot_lq: ./datasets/RealSet65
      mean: 0.5
      std: 0.5
      io_backend:
        type: disk
  
  val_3:
    name: RealSR
    type: PairedImageDataset
    # dataroot_gt: datasets/TestDataSR/HR/B100/x2
    dataroot_lq: ./datasets/RealSR/lq
    dataroot_gt: ./datasets/RealSR/gt
    mean: 0.5
    std: 0.5
    io_backend:
      type: disk
  

# network structures
network_g:
  type: EDMUNet
  img_resolution: 64
  img_channels: 3
  label_dim: 0
  augment_dim: 0
  model_channels: 192
  use_fp16: False
  model_type: 'SongUNet'
  embedding_type: 'positional'
  fourier_scale: 8
  encoder_type: 'standard'
  decoder_type: 'standard'
  channel_mult_noise: 1
  channel_mult_emb: 4
  resample_filter: [1, 1,]
  channel_mult: [1, 2, 2, 2,]
  attn_resolutions: [64, 32, 16, 8]
  dropout: 0.13
  sigma_data: 0.5
  sigma_min: 0
  cond_lq: True
  swin: True
  scale: 4
  use_enc: True
  upsampler: 'nearest'
  res_con: False
  swin_params:
    embed_dim: 192
    use_conv: True
    num_heads: 6
    window_size: 8
    depth: 1
    patch_size: 1
    mlp_ratio: 4
    qkv_bias: True
    qk_scale: ~
    drop: 0
    attn_drop: 0.
    drop_path: 0.
    use_checkpoint: False
    patch_norm: False

# path
path:
  pretrain_network_g: /root/CTMSR.pth  #Path of pretrained model
  strict_load_g: true
  param_key_g: params_ema
# training settings
train:
  consistency_opt:
    sigma_min: 0
    sigma_max: 1

# validation settings
val:
  suffix: ~
  save_img: true
  chop_size: 512
  chop_stride: 448
  min_size: 64

  metrics:
    psnr: # metric name, can be arbitrary
      type: calculate_psnr
      crop_border: 0
      test_y_channel: yes
    
    ssim: # metric name, can be arbitrary
      type: calculate_ssim
      crop_border: 0
      test_y_channel: yes

    lpips:
      type: calculate_lpips
      better: lower
      
    clipiqa:
      type: calculate_clipiqa
      better: higher
    
    musiq:
      type: calculate_musiq
      better: higher
    
    maniqa:
      type: calculate_maniqa
      better: higher
    
    niqe:
      type: calculate_niqe
      better: lower

<img width="381" height="380" alt="Image" src="https://github.com/user-attachments/assets/94b1d6d4-1a9a-4cf2-ba8f-87cbfdea502f" />

**Comment by meimeizeng** [comment]
是的，我的低分辨率图像是通过bicubic下采样得到的，分辨率大小为64*64，正是取自于ImageNet-Test测试集，我刚刚重新测试，仍然会遇到这个问题，下面为ground truth和推理图像的对比。我也是使用的链接里发布的预训练模型。

<img width="216" height="217" alt="Image" src="https://github.com/user-attachments/assets/c133cd4a-dc81-4c23-b74b-898af84c020e" />

<img width="315" height="314" alt="Image" src="https://github.com/user-attachments/assets/4a06e71c-9c7e-4963-880e-545f640317c8" />

**Comment by meimeizeng** [comment]
感谢作者，得到您的确认后，我进行了全面的检查，考虑到应该是测试时修改了模型设置，部分模型未能加载，当我重新加载所有模型，可以得到正确，清晰的推理结果。谢谢作者的耐心回复！

<img width="194" height="191" alt="Image" src="https://github.com/user-attachments/assets/8cd7546f-bc44-4dd4-b0a7-b233e85adabb" />
